src/sentry/workflow_engine/models/data_condition.py

```diff
@@ -57,8 +57,24 @@ class DataCondition(DefaultFieldsModel):
         on_delete=models.CASCADE,
     )
 
-    def evaluate_value(self, value: float | int) -> DetectorPriorityLevel | None:
-        # Note: We'll have other types other than int/DetectorPriorityLevel here, keeping it simple for now
+    def get_condition_result(self) -> DataConditionResult:
+        match self.condition_result:
+            case float() | bool():
```

Comment:
I haven't worked with these patterns much yet, but does this basically do `float(self.condition_result) | bool(self.condition_result)`?

If `condition_result` was a string like `test`, would this throw an exception?

Reply:
The way this is working is to use [`class patterns`](https://docs.python.org/3/reference/compound_stmts.html#class-patterns) -- it's doing a few different kinds of checks under the hood.

there's a test case setup for a string, but basically what happens is that it tries to evaluate all the types and then falls through to the default `_` case and will log an error.